oauth2_http/java/com/google/auth/oauth2/Slf4jLoggingHelpers.java

```diff
@@ -0,0 +1,180 @@
+            Map<String, Object> contextMap = parseGenericData(data);
+            loggingDataMap.put("request.payload", gson.toJson(contextMap));
+          } else if (request.getContent() instanceof JsonHttpContent) {
+            String jsonData = gson.toJson(((JsonHttpContent) request.getContent()).getData());
+            loggingDataMap.put("request.payload", jsonData);
+          }
+
+          Slf4jUtils.log(logger, org.slf4j.event.Level.DEBUG, loggingDataMap, message);
+        } else {
+
+          Slf4jUtils.log(logger, org.slf4j.event.Level.INFO, loggingDataMap, message);
+        }
+      }
+    } catch (Exception e) {
+      // let logging fail silently
+    }
+  }
+
+  static void logResponse(HttpResponse response, LoggerProvider loggerProvider, String message) {
+    try {
+      Logger logger = loggerProvider.getLogger();
+      if (logger.isInfoEnabled()) {
+        Map<String, Object> responseLogDataMap = new HashMap<>();
+        responseLogDataMap.put("response.status", String.valueOf(response.getStatusCode()));
+        responseLogDataMap.put("response.status.message", response.getStatusMessage());
+
+        Map<String, Object> headers = new HashMap<>(response.getHeaders());
+        responseLogDataMap.put("response.headers", headers.toString());
+        Slf4jUtils.log(logger, org.slf4j.event.Level.INFO, responseLogDataMap, message);
+      }
+    } catch (Exception e) {
+      // let logging fail silently
+    }
+  }
+
+  static void logResponsePayload(
+      GenericData genericData, LoggerProvider loggerProvider, String message) {
+    try {
+
+      Logger logger = loggerProvider.getLogger();
+      if (logger.isDebugEnabled()) {
+        Map<String, Object> contextMap = parseGenericData(genericData);
+        Slf4jUtils.log(logger, org.slf4j.event.Level.DEBUG, contextMap, message);
+      }
+    } catch (Exception e) {
+      // let logging fail silently
+    }
+  }
+
+  private static Map<String, Object> parseGenericData(GenericData genericData) {
+    Map<String, Object> contextMap = new HashMap<>();
+    genericData.forEach(
+        (key, val) -> {
+          if (SENSITIVE_KEYS.contains(key)) {
+            String secretString = String.valueOf(val);
+            String hashedVal = calculateSHA256Hash(secretString);
+            contextMap.put(key, hashedVal);
+          } else {
+            contextMap.put(key, val.toString());
+          }
+        });
+    return contextMap;
+  }
+
+  private static String calculateSHA256Hash(String data) {
+    try {
+      MessageDigest digest = MessageDigest.getInstance("SHA-256");
+      byte[] inputBytes = data.getBytes(StandardCharsets.UTF_8);
+      byte[] hashBytes = digest.digest(inputBytes);
+      return bytesToHex(hashBytes);
+    } catch (NoSuchAlgorithmException e) {
+      return "Error calculating SHA-256 hash."; // do not fail for logging failures
+    }
+  }
+
+  private static String bytesToHex(byte[] hash) {
```

Comment:
Could you add a blurb to generalize what this is doing. Converting this to a hex value and XOR's 0xff (255?) with each byte because of reason X

Reply:
[e1d9a78](https://github.com/googleapis/google-auth-library-java/pull/1586/commits/e1d9a78d76f037690e87a7fbbe8314f0b4f3cbf8)